pkg/operation/botanist/operatingsystemconfig.go

```diff
@@ -164,13 +164,27 @@ func (b *Botanist) deployOperatingSystemConfigsForWorker(ctx context.Context, ma
 
 	sshKey := b.Secrets[v1beta1constants.SecretNameSSHKeyPair].Data[secrets.DataKeySSHAuthorizedKeys]
 
+	criNamesConfig := map[string]interface{}{
+		"containerd": extensionsv1alpha1.CRINameContainerD,
+	}
+
+	workerNameLabel := map[string]interface{}{
+		"name": extensionsv1alpha1.CRINameWorkerLabel,
+	}
+
+	criConfig := map[string]interface{}{
+		"names":  criNamesConfig,
+		"labels": workerNameLabel,
+	}
+
 	originalConfig["osc"] = map[string]interface{}{
 		"type":                 machineImage.Name,
 		"purpose":              extensionsv1alpha1.OperatingSystemConfigPurposeReconcile,
 		"reloadConfigFilePath": common.CloudConfigFilePath,
 		"secretName":           secretName,
 		"customization":        customization,
 		"sshKey":               string(sshKey),
+		"criConfig":            criConfig,
```

Comment:
i think this should be:
```suggestion
		"cri":            criConfig,
```
[see here](https://github.com/gardener/gardener/blob/ba17ac74b277e3e2b9403d85ea9509fb2005b65f/charts/seed-operatingsystemconfig/original/templates/node/_node-cri.labels#L2)